bitcoin/src/blockdata/fee_rate.rs

```diff
@@ -0,0 +1,158 @@
+//! Implements `FeeRate` and assoctiated features.
+
+use core::fmt;
+use core::ops::{Mul, MulAssign, Div, DivAssign};
+use crate::Amount;
+use super::Weight;
+
+/// Represents fee rate.
+///
+/// This is an integer newtype representing fee rate in `sat/kwu`. It provides protection against mixing
+/// up the types as well as basic formatting features.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+#[cfg_attr(feature = "serde", serde(transparent))]
+pub struct FeeRate(u64);
```

Comment:
Integer (aka "round") FeeRate makes me wary. Any time you combine two PSBTs the resulting feerate is likely a non-integer rational. 

It's a privacy concern from the payjoin spec:
> If the sender's fee rate is always round, then a blockchain analyst can easily spot the transactions of the sender involving payjoin by checking if, when removing a single input to the suspected payjoin transaction, the resulting fee rate is round. To prevent this, the sender can agree to pay more fee so the receiver make sure that the payjoin transaction fee is also round.


Could representing it as FeeRate(Amount in sats / Weight) fix this problem?

Reply:
It's in sat/kwu, same as Core uses and I don't think any wallet uses anything more precise. Even floats have limits on precision, so there's no way to solve it perfectly.

I think for PayJoin we should check if the fee rate exactly matches sat/vb and if yes, match it, if not we should check sat/kwu. The protocol already fixes it so we don't have to worry about it here.

Reply:
I didn't realize it was sat/kwu. Makes sense to me. Don't reinvent the wheel.